doc/parsing-overview.md

```diff
@@ -0,0 +1,227 @@
+
+```csharp
+var markdownText = File.ReadAllText("sample.md");
+
+// No pipeline provided means a default pipeline will be used
+var document = MarkdownParser.Parse(markdownText);
+```
+
+Pipelines can be created and configured manually, however this must be done using a `MarkdownPipelineBuilder` object, which then is configured through a fluent interface composed of extension methods.  
+
+```csharp
+var markdownText = File.ReadAllText("sample.md");
+
+// Markdig's "UseAdvancedExtensions" option includes many common extensions beyond
+// CommonMark, such as citations, figures, footnotes, grid tables, mathematics
+// task lists, diagrams, and more.
+var pipeline = new MarkdownPipelineBuilder()
+    .UseAdvancedExtensions()
+    .Build();
+
+var document = MarkdownParser.Parse(markdownText, pipeline);
+```
+
+Extensions can also be added individually:
+
+```csharp
+var markdownText = File.ReadAllText("sample.md");
+
+var pipeline = new MarkdownPipelineBuilder()
+    .UseCitations()
+    .UseFootnotes()
+    .UseMyCustomExtension()
+    .Build();
+
+var document = MarkdownParser.Parse(markdownText, pipeline);
+```
+
+## The Parser and the Pipeline
+
+As metioned in the [Introduction](#introduction), Markdig's parsing machinery involves two surface components: the `MarkdownParser`, and the `MarkdownPipeline`.  The main parsing algorithm (not to be confused with individual `BlockParser` and `InlineParser` components) lives in the `MarkdownParser.Parse(...)` static method. The `MarkdownPipeline` is responsible for configuring the behavior of the parser.
+
+These two components are covered in further detail in the following sections.
+
+### The MarkdownPipeline
+
+The `MarkdownPipeline` is a sealed internal class which dictates what features the parsing algorithm has.  The pipeline must be created by using a `MarkdownPipelineBuilder` as shown in the examples above.
+
+The `MarkdownPipeline` holds configuration information and collections of extensions and parsers.  Parsers fall into one of two categories:
+
+* Block Parsers (`BlockParser`)
+* Inline Parsers (`InlineParser`)
+
+Extensions are classes implementing `IMarkdownExtension` which are allowed to add to the list of parsers, or modify existing parsers and/or renderers. They are invoked to perform their mutations on the pipeline when the pipeline is built by the `MarkdownPipelineBuilder`.
+
+Lastly, the `MarkdownPipeline` contains a few extra elements:
+
+* A configuration setting determining whether or not trivial elements, referred to as *trivia*, (whitespace, extra heading characters, unescaped strings, etc) are to be tracked
+* A configuration setting determining whether or not nodes in the resultant abstract syntax tree will refer to their precise original locations in the source
+* An optional delegate which will be invoked when the document has been processed.
+* An optional `TextWriter` which will get debug logging from the parser
```

Comment:
Worth noting that `Markdig` itself won't add anything to this log, but an extension author may choose to use it when debugging their implementation.